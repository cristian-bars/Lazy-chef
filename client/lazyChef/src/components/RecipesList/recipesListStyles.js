import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#D3E0FF',
  },
  logo: {
    height: 150,
  },
  imageIcons: {
    width: 50,
    height: 50,
  },
  recipeDetail: {
    flexDirection: 'column',
    backgroundColor: 'white',
    margin: 10,
  },
  titleText: {
    fontSize: 20,
    flex: 1,
    padding: 10,
    flexWrap: 'wrap',
  },
  descriptionText: {
    fontSize: 15,
    flexWrap: 'wrap',
    padding: 10,
  },
  recipeIcons: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  input: {
    width: 360,
    marginTop: 20,
    height: 50,
    paddingLeft: 15,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    textAlign: 'center',
    fontSize: 25,
    margin: 15,
    borderWidth: 2,
    borderRadius: 25,
    borderColor: '#397ED0',
    backgroundColor: 'white',
  },
  inputText: {
    paddingLeft: 20,
    fontSize: 20,
  },
  searchImage: {
    width: 30,
    height: 30,
    opacity: 0.7,
  },
  ingList: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  ingredientsList: {
    marginHorizontal: 10,
    backgroundColor: 'white',
    fontSize: 30,
  },
  wordsTitle: {
    fontSize: 30,
    color: '#FF914D',
    backgroundColor: 'white',
  },
  optionslist: {
    flexWrap: 'wrap',
  },
});

export default styles;
