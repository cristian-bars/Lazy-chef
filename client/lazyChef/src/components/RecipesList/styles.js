import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#D3E0FF',
  },
  ingList: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingVertical: 5,
  },
  ingredientsList: {
    marginHorizontal: 10,
    backgroundColor: 'white',
    fontSize: 25,
    borderRadius: 5,
    alignContent: 'flex-end',
  },
  wordsTitle: {
    fontSize: 25,
    color: '#FF914D',
    backgroundColor: 'white',
    borderRadius: 5,
    marginLeft: 10,
  },
  optionslist: {
    flexWrap: 'wrap',
  },
});

export default styles;
