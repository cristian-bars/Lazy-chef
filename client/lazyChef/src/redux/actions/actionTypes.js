export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
