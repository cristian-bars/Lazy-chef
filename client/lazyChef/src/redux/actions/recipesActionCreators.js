import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
