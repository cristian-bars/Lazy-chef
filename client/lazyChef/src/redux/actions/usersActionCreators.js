import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
