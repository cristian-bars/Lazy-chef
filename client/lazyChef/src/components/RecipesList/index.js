/* eslint-disable react-hooks/exhaustive-deps */
import React, {useEffect} from 'react';
import {View, Text, FlatList, Image, TouchableOpacity} from 'react-native';
import {connect} from 'react-redux';
import {PropTypes} from 'prop-types';
import {loadRecipes} from '../../redux/actions/recipesActionCreators';
import styles from './styles';
import generalStyles from '../../../generalStyles';

const RecipesList = ({recipes, dispatch, route, navigation}) => {
  const {ingredient} = route.params;

  useEffect(() => {
    if (!recipes.length) {
      dispatch(loadRecipes());
    }
  }, [recipes]);

  const listRender = ({item}) => {
    return (
      <TouchableOpacity
        style={generalStyles.recipeDetail}
        onPress={() => navigation.navigate('Detail', {recipeId: item._id})}>
        {item.image[0] ? (
          <Image
            style={generalStyles.recipeLogo}
            source={{uri: item.image[0]}}
          />
        ) : (
          <Image
            style={generalStyles.recipeLogo}
            source={{
              uri: 'https://i.ibb.co/PwFdwdH/Vector-artistic-pen-and-ink-drawing-illustration-of-empty-plate-knife-and-fork.jpg',
            }}
          />
        )}

        <Text style={generalStyles.titleText}>{item.title}</Text>
        <View style={generalStyles.recipeIcons}>
          <View>
            <Image
              style={generalStyles.imageIcons}
              source={require('../../img/Ingredientes.jpg')}
            />
            <Text style={generalStyles.descriptionText}>
              {item.recipeIngredient.length}
            </Text>
          </View>
          <View>
            <Image
              style={generalStyles.imageIcons}
              source={require('../../img/herramientas-y-utensilios.png')}
            />
            <Text style={generalStyles.descriptionText}>{item.difficulty}</Text>
          </View>
          <View>
            <Image
              style={generalStyles.imageIcons}
              source={require('../../img/duration.png')}
            />
            <Text style={generalStyles.descriptionText}>{item.totalTime}</Text>
          </View>
        </View>

        <Text style={generalStyles.descriptionText}>{item.Description}</Text>
      </TouchableOpacity>
    );
  };

  const ingredientRender = ({item}) => {
    return <Text style={styles.ingredientsList}>{item}</Text>;
  };
  return (
    <View style={styles.container}>
      <View style={styles.ingList}>
        <Text style={styles.wordsTitle}>Recetas con </Text>
        <FlatList
          style={styles.optionslist}
          data={ingredient}
          horizontal={true}
          keyExtractor={item => item}
          renderItem={ingredientRender}
        />
      </View>
      <View>
        {recipes.length ? (
          <FlatList
            style={styles.list}
            data={recipes}
            keyExtractor={item => item._id}
            renderItem={listRender}
          />
        ) : (
          <Text>No tenim receptes</Text>
        )}
      </View>
    </View>
  );
};

RecipesList.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(store) {
  return {
    recipes: store.recipes,
  };
}

export default connect(mapStateToProps)(RecipesList);
