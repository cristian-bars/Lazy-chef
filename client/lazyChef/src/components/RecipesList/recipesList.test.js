import 'react-native';
import React from 'react';
import RecipesList from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/recipesActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a RecipesList component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  let route;
  beforeEach(() => {
    jest.spyOn(actions, 'loadRecipes').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      recipes: [
        {
          title: 'macarrones',
          description: '1a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
        {
          title: 'Queso',
          description: '2a',
          image: ['a'],
          recipeIngredient: ['b'],
        },
      ],
    });
    navigation = {
      navigate: jest.fn(),
    };
    route = {
      params: {
        ingredient: 'tomate',
      },
    };
  });

  test('renders correctly', () => {
    const wrapper = render(
      <Provider store={myStore}>
        <RecipesList route={route} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('When recipe is pressed', () => {
    test('Then navigation.navigate is invoked to show recipe detail', () => {
      myStore = mockStore({
        recipes: [
          {
            title: 'macarrones',
            description: '1a',
            image: ['a'],
            recipeIngredient: ['b'],
          },
        ],
      });
      const {getByTestId} = render(
        <Provider store={myStore}>
          <RecipesList route={route} navigation={navigation} />
        </Provider>,
      );
      const recipeDetail = getByTestId('moveToDetail');
      fireEvent.press(recipeDetail);

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
