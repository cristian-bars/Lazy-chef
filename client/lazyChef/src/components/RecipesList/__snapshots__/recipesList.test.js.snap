// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RecipesList component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#D3E0FF",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "color": "#FF914D",
          "fontSize": 25,
        }
      }
    >
      Recetas con 
    </Text>
    <RCTScrollView
      data="tomate"
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flexWrap": "wrap",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "alignContent": "flex-end",
                "backgroundColor": "white",
                "borderRadius": 5,
                "fontSize": 25,
                "marginHorizontal": 10,
              }
            }
          >
            t
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "alignContent": "flex-end",
                "backgroundColor": "white",
                "borderRadius": 5,
                "fontSize": 25,
                "marginHorizontal": 10,
              }
            }
          >
            o
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "alignContent": "flex-end",
                "backgroundColor": "white",
                "borderRadius": 5,
                "fontSize": 25,
                "marginHorizontal": 10,
              }
            }
          >
            m
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "alignContent": "flex-end",
                "backgroundColor": "white",
                "borderRadius": 5,
                "fontSize": 25,
                "marginHorizontal": 10,
              }
            }
          >
            a
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "alignContent": "flex-end",
                "backgroundColor": "white",
                "borderRadius": 5,
                "fontSize": 25,
                "marginHorizontal": 10,
              }
            }
          >
            t
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <Text
            style={
              Object {
                "alignContent": "flex-end",
                "backgroundColor": "white",
                "borderRadius": 5,
                "fontSize": 25,
                "marginHorizontal": 10,
              }
            }
          >
            e
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View>
    <RCTScrollView
      data={
        Array [
          Object {
            "description": "1a",
            "image": Array [
              "a",
            ],
            "recipeIngredient": Array [
              "b",
            ],
            "title": "macarrones",
          },
          Object {
            "description": "2a",
            "image": Array [
              "a",
            ],
            "recipeIngredient": Array [
              "b",
            ],
            "title": "Queso",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "flexDirection": "column",
                "margin": 10,
                "opacity": 1,
              }
            }
            testID="moveToDetail"
          >
            <Image
              source={
                Object {
                  "uri": "a",
                }
              }
              style={
                Object {
                  "height": 150,
                }
              }
            />
            <Text
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "padding": 10,
                }
              }
            >
              macarrones
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/Ingredientes.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                >
                  1
                </Text>
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/herramientas-y-utensilios.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/duration.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 15,
                  "padding": 10,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "flexDirection": "column",
                "margin": 10,
                "opacity": 1,
              }
            }
            testID="moveToDetail"
          >
            <Image
              source={
                Object {
                  "uri": "a",
                }
              }
              style={
                Object {
                  "height": 150,
                }
              }
            />
            <Text
              style={
                Object {
                  "flex": 1,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "padding": 10,
                }
              }
            >
              Queso
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                }
              }
            >
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/Ingredientes.jpg",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                >
                  1
                </Text>
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/herramientas-y-utensilios.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/img/duration.png",
                    }
                  }
                  style={
                    Object {
                      "height": 50,
                      "width": 50,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "flexWrap": "wrap",
                      "fontSize": 15,
                      "padding": 10,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 15,
                  "padding": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
